### PR TITLE
Adding TF support for Tenancy APIs for Membership RBAC Role Bindings

### PR DESCRIPTION
**Pull request by sahsagar-google**
Fixes https://github.com/hashicorp/terraform-provider-google/issues/15167

If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [X] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [X] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [X] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/run-provider-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read [Write release notes](https://googlecloudplatform.github.io/magic-modules/contribute/release-notes/) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:new-resource
`google_gke_hub_membership_rbac_role_binding`
```

### CODE REVIEW

File: mmv1/products/gkehub2/MembershipRBACRoleBinding.yaml [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+--- !ruby/object:Api::Resource
+name: 'MembershipRBACRoleBinding'
+base_url: 'projects/{{project}}/locations/{{location}}/memberships/{{membership_id}}/rbacrolebindings'
+create_url: 'projects/{{project}}/locations/{{location}}/memberships/{{membership_id}}/rbacrolebindings/?rbacrolebinding_id={{membership_rbac_role_binding_id}}'
+update_url: 'projects/{{project}}/locations/{{location}}/memberships/{{membership_id}}/rbacrolebindings/{{membership_rbac_role_binding_id}}'
```
Comment (ScottSuarez) [inline mmv1/products/gkehub2/MembershipRBACRoleBinding.yaml:18]: Could you add a handwritten update test for this resource?
https://googlecloudplatform.github.io/magic-modules/develop/add-handwritten-test/#update-tests

File: mmv1/templates/terraform/examples/gkehub_membership_rbac_role_binding_basic.tf.erb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    }
+  }
+  
+  depends_on = [google_container_cluster.primary]
+}
+
+resource "google_gke_hub_membership_rbac_role_binding" "<%= ctx[:primary_resource_id] %>" {
+  membership_rbac_role_binding_id = "tf-test-membership-rbac-role-binding%{random_suffix}"
+  membership_id = "tf-test-membership%{random_suffix}"
+  user = "service-155926358489@gcp-sa-anthossupport.iam.gserviceaccount.com"
```
Comment (ScottSuarez) [inline mmv1/templates/terraform/examples/gkehub_membership_rbac_role_binding_basic.tf.erb:21]: Is this value stable? I worry about an external dependency here.
```
  user = "service-155926358489@gcp-sa-anthossupport.iam.gserviceaccount.com"
  ```

File: mmv1/products/gkehub2/MembershipRBACRoleBinding.yaml [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+      - group
+  - !ruby/object:Api::Type::String
+    name: 'group'
+    description: |
+      Principal that is be authorized in the cluster (at least of one the oneof
+      is required). Updating one will unset the other automatically.
+      group is the group, as seen by the kubernetes cluster.
+    exactly_one_of:
+      - user
+      - group
```
Comment (ScottSuarez) [inline mmv1/products/gkehub2/MembershipRBACRoleBinding.yaml:145]: Could you set this value in a test?

File: mmv1/products/gkehub2/MembershipRBACRoleBinding.yaml [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+    name: 'role'
+    description: |
+      Role to bind to the principal.
+    required: true
+    properties:
+      - !ruby/object:Api::Type::Enum
+        name: 'predefinedRole'
+        description: |
+          PredefinedRole is an ENUM representation of the default Kubernetes Roles
+        values:
```
Comment (ScottSuarez) [inline mmv1/products/gkehub2/MembershipRBACRoleBinding.yaml:156]: Can this property be required so an empty block isn't possible ?

**Review by ScottSuarez — CHANGES_REQUESTED** [review]

File: mmv1/products/gkehub2/MembershipRBACRoleBinding.yaml (hunk 3 above)
Comment (sahsagar-google) [reply mmv1/products/gkehub2/MembershipRBACRoleBinding.yaml:145]: Removing this field; recently found out its not usable in the API.

File: mmv1/templates/terraform/examples/gkehub_membership_rbac_role_binding_basic.tf.erb (hunk 2 above)
Comment (sahsagar-google) [reply mmv1/templates/terraform/examples/gkehub_membership_rbac_role_binding_basic.tf.erb:21]: That's a great catch, thank you.

File: mmv1/products/gkehub2/MembershipRBACRoleBinding.yaml (hunk 1 above)
Comment (sahsagar-google) [reply mmv1/products/gkehub2/MembershipRBACRoleBinding.yaml:18]: Misunderstood the resource to be updatable, but it's not. Removed update logic and made the resource immutable.

File: mmv1/templates/terraform/examples/gkehub_membership_rbac_role_binding_basic.tf.erb (hunk 2 above)
Comment (sahsagar-google) [reply mmv1/templates/terraform/examples/gkehub_membership_rbac_role_binding_basic.tf.erb:21]: I have asked a YAQS question for this, and still figuring out how to fix this. Thank you

Comment (sahsagar-google) [reply mmv1/templates/terraform/examples/gkehub_membership_rbac_role_binding_basic.tf.erb:21]: https://yaqs.corp.google.com/eng/q/7914452922010697728

Comment (sahsagar-google) [reply mmv1/templates/terraform/examples/gkehub_membership_rbac_role_binding_basic.tf.erb]: Made a change. Please review it. Thank you

**Review by ScottSuarez — APPROVED** [review]

**Review by ScottSuarez — CHANGES_REQUESTED** [review]
Could you pull the latest changes from `main`? 

Looks like there are some failures in the testing pipeline due to an old base

**Review by ScottSuarez — APPROVED** [review]

**Comment by ScottSuarez** [comment]
Notice: it appears this resource is `beta` only.. Causing test failures in our GA pipeline. Please mark resources min_version:beta in the future. Need to hotfix for release.

**Pull request merged**